### DISCUSSION TITLE
Could not poll the database: SQLSTATE[HY000] [2002] No such file or directory

### DISCUSSION THREAD

**Discussion by mrtargaryen** [firefly-iii/firefly-iii#6987] [Firefly III]
I just install fireflyiii on a Raspberry Pi 3B in docker and went to 192.168.50.99:80 in my browser and received this message:

`Could not poll the database: SQLSTATE[HY000] [2002] No such file or directory (SQL: select `id`, `name`, `data` from `configuration` where `name` = is_demo_site and `configuration`.`deleted_at` is null limit 1) 
`

And this in the log: 

`SQLSTATE[HY000] [2002] No such file or directory (SQL: insert into `oauth_clients` (`user_id`, `name`, `secret`, `provider`, `redirect`, `personal_access_client`, `password_client`, `revoked`, `updated_at`, `created_at`) values (?, Firefly III Personal Access Client, qwrVPF9JOpPVEfD8OIc4rBCWqkcmDEsse9xWR8Kr, ?, http://localhost, 1, 0, 0, 2023-02-07 15:38:36, 2023-02-07 15:38:36))`

I'm only new to Linux, so I'm not too sure what that means. Any ideas?

**Comment by JC5** [comment]
https://github.com/orgs/firefly-iii/discussions?discussions_q=%22no+such+file+or+directory%22

**Reply by mrtargaryen** [reply]
That doesn't really answer anything, I'm not sure what to do. I have MariaDB installed. Do I create the DB first? Then add it to the docker run command?

**Reply by JC5** [reply]
Try using the Docker Compose file, that may be easier. https://docs.firefly-iii.org/firefly-iii/installation/docker/

**Reply by mrtargaryen** [reply]
> Try using the Docker Compose file, that may be easier. https://docs.firefly-iii.org/firefly-iii/installation/docker/

That did the trick, I created the docker-compose.yml, and used the example .env file (made changes to it) and it worked perfectly. Thanks.

**Comment by SethranKada** [comment]
I've been having the same issue, and no matter what I do I always run into this same issue over and over again. Has anyone figured out how to fix it?

**Reply by JC5** [reply]
What have you done so far? What system are you running?

**Reply by SethranKada** [reply]
I seem to have fixed the issue through random tinkering. Seems the issue is related to the networking between containers. I solved it by running MariaDB natively on my Synology NAS and mounting the mysqld.sock in the firefly container. It took nearly half an hour for the initial setup, which was surprising, but it ended up working in the end.

For anyone else with this issue, I recommend experimenting with your docker networking settings and seeing how the error message changes. Even while still using docker for the DB I was able to change the error message from 'No such file or directory' to 'Permission Denied'. Which was much closer to working.
